crates/node/src/kitchensink.rs

```diff
@@ -0,0 +1,618 @@
+    process_substrate: Option<Child>,
+    process_proxy: Option<Child>,
+}
+
+impl KitchensinkNode {
+    const SUBSTRATE_READY_MARKER: &str = "Running JSON-RPC server";
+    const ETH_PROXY_READY_MARKER: &str = "Running JSON-RPC server";
+    const BASE_SUBSTRATE_RPC_PORT: u16 = 9944;
+    const BASE_PROXY_RPC_PORT: u16 = 8545;
+
+    fn spawn_process(&mut self, genesis: String) -> anyhow::Result<()> {
+        let substrate_rpc_port = Self::BASE_SUBSTRATE_RPC_PORT + self.id as u16;
+        let proxy_rpc_port = Self::BASE_PROXY_RPC_PORT + self.id as u16;
+
+        self.rpc_url = format!("http://127.0.0.1:{proxy_rpc_port}");
+
+        let chainspec_path = self.generate_chainspec_file(&genesis)?;
+
+        // Start Substrate node
+        let mut substrate_process = Command::new(&self.substrate_binary)
+            .arg("--chain")
+            .arg(chainspec_path)
+            .arg("--rpc-port")
+            .arg(substrate_rpc_port.to_string())
+            .arg("--name")
+            .arg(format!("revive-kitchensink-{}", self.id))
+            .arg("--rpc-methods")
+            .arg("Unsafe")
+            .env(
+                "RUST_LOG",
+                "error,evm=debug,sc_rpc_server=info,runtime::revive=debug",
+            )
+            .stdout(Stdio::null())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        // Give the node a moment to boot
+        Self::wait_ready(
+            &mut substrate_process,
+            Self::SUBSTRATE_READY_MARKER,
+            Duration::from_secs(30),
+        )?;
+
+        let mut proxy_process = Command::new(&self.eth_proxy_binary)
+            .arg("--dev")
+            .arg("--rpc-port")
+            .arg(proxy_rpc_port.to_string())
+            .arg("--node-rpc-url")
+            .arg(format!("ws://127.0.0.1:{substrate_rpc_port}"))
+            .env("RUST_LOG", "info,eth-rpc=debug")
+            .stdout(Stdio::null())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        Self::wait_ready(
+            &mut proxy_process,
+            Self::ETH_PROXY_READY_MARKER,
+            Duration::from_secs(30),
+        )?;
+
+        self.process_substrate = Some(substrate_process);
+        self.process_proxy = Some(proxy_process);
+
+        Ok(())
+    }
+
+    fn generate_chainspec_file(&self, genesis_path: &str) -> anyhow::Result<PathBuf> {
+        let mut merged_balances: Vec<(String, u128)> = vec![
+            (
+                "5Ff3iXP75ruzroPWRP2FYBHWnmGGBSb63857BgnzCoXNxfPo".to_string(),
+                1_000_000_000_000_000_000_000,
+            ),
+            (
+                "5Fbsd6WXDGiLTxunqeK5BATNiocfCqu9bS1yArVjCgeBLkVy".to_string(),
+                1_000_000_000_000_000_000_000,
+            ),
+        ];
+        let mut eth_balances = self.extract_balance_from_genesis_file(genesis_path)?;
+        merged_balances.append(&mut eth_balances);
+
+        let chainspec: ChainSpec = serde_json::from_str(include_str!("default_chainspec.json"))?;
```

Comment:
To move forward, let's just resort back to your original untyped approach. So just reading it into a `Value`, change the endowed and mapped accounts and supply the genesis config it to the builder.